src/frontend/hooks/goose-form.js

```diff
@@ -80,74 +80,86 @@ const useGooseForm = () => {
   }, [history]);
 
   const saveForm = useCallback(async () => {
-    // TODO: Validation checks here.
 
-    // Send data to server:
     try {
-      const files = images;
-      let imageUrls = [];
-
-      if (files.length > 0) {
+      // image files added 
+      const newlyAddedImages = images;
+  
+      // all image files uploaded 
+      let newlyAddedImageUrls = [];
+      
+      if (newlyAddedImages.length > 0) {
+        
         const data = new FormData();
 
-        files.forEach((file) => {
+        newlyAddedImages.forEach((file) => {
           data.append('files', file, file.name);
         });
 
         const res = await api.formUpload(data);
 
-        imageUrls = res.data.data;
-        if (imageUrls.length === 0) {
+        // array of added image urls
+        newlyAddedImageUrls = res.data.data;
+
+        if (newlyAddedImageUrls.length === 0) {
           throw new Error('Failed to upload image.');
         }
       }
-
+    
       const gooseInfo = {
         name: gooseName,
         description,
         updatedAt: Date.now(),
       };
-
-      if (gooseName && description && imageUrls.length > 0) {
+      
+      // if fields are non-empty and if there is a preexisting image or a new one added
+      if (gooseName && description && (newlyAddedImageUrls.length > 0 || imageUrls.length > 0)) {
         if (params.gooseId) {
           await api.geeseInfo.updateGeeseInfo(params.gooseId, gooseInfo);
           await Promise.all(
             imagesToDelete.map((imageId) => {
               return api.geeseInfo.deleteGeeseImages(imageId);
             }),
           );
+
+          // upload images if new files are added
+          if (newlyAddedImages.length >0){
           await api.geeseInfo.addGeeseImages(
-            imageUrls.map((image) => {
+            newlyAddedImageUrls.map((image) => {
               return {
                 gooseId: params.gooseId,
                 imgDir: image,
               };
             }),
           );
-        } else {
-          // // TODO when new goose is implemented
+          }
+        } 
+        else {
           const res = await api.geeseInfo.addGeeseInfo(gooseInfo);
           await api.geeseInfo.addGeeseImages(
-            imageUrls.map((image) => {
+            newl.map((image) => {
```

Comment:
should this be `newlyAddedImageUrls`

Reply:
> should this be `newlyAddedImageUrls`

whoops - good catch! - must've accidentally fat-fingered cntl-z'd before pushing.. :P